packages/engine/src/scene/behaviors/setReflectionProbe.ts

```diff
@@ -39,25 +36,32 @@ export const setReflectionProbe: Behavior = (entity, args: {}) => {
     }
 
 
-      Engine.scene.traverse(child=>{
-        const mat=(child as any).material;
-        if(mat){
-          mat.envMapIntensity=options.intensity;
-          mat.onBeforeCompile = function ( shader ) {
-              shader.uniforms.cubeMapSize = { value: options.probeScale};
-              shader.uniforms.cubeMapPos = { value: options.probePositionOffset};
-              shader.vertexShader = 'varying vec3 vWorldPosition;\n' + shader.vertexShader;
-              shader.vertexShader = shader.vertexShader.replace(
-                  '#include <worldpos_vertex>',
-                  worldposReplace
-              );
-              shader.fragmentShader = shader.fragmentShader.replace(
-                  '#include <envmap_physical_pars_fragment>',
-                  envmapPhysicalParsReplace
-              );
-        };
-      }
-    });
-  
+      Engine.entities.forEach(entity=>{
```

Comment:
This would be quite expensive on big scenes, we should push this to an added query for Object3DComponent in a system.

Reply:
ok on it